Update for signac 2.0 and general cleanup (#183)
glotzerlab/signac-docs: docs/source/flow-project.rst

==========

It is highly recommended to divide individual modifications of your project's data space into distinct functions.

> In this context, an *operation* is defined as a function whose only positional arguments are instances of :py:class:`~signac.contrib.job.Job`.

Reviewer: ```suggestion
In this context, an *operation* is defined as a function whose only positional arguments are instances of :py:class:`~signac.job.Job`.
```

    if __name__ == "__main__":
        MyProject().main()

.. tip::

>     By default operations only act on a single job and can simply be defined with the signature `def op(job)`.

Reviewer 2: Use code font rather than italics.
```suggestion
    By default operations only act on a single job and can simply be defined with the signature ``def op(job)``.
```

>     When using :ref:`aggregate operations <aggregation>`, it is recommended to define operations as accepting a variadic list of ``*jobs`` parameters so that the operation is not restricted to a specific aggregate size.

Reviewer: Need to define variadic

Reviewer 2: "Variadic" is the right word to use, it's a term used in many programming languages. I think it's also self-explanatory because it's pointing clearly to the `*jobs` argument. It might be a little awkward to link to a blog post explaining how `*args` work but we could do something along those lines. @cbkerr How would you resolve this?

Reviewer: ```suggestion
    When using :ref:`aggregate operations <aggregation>`, it is recommended to allow the operation to accept a variable number of jobs using a variadic argument ``*jobs``, so that the operation is not restricted to a specific aggregate size.
```

    for more information.

We can define both :py:meth:`~flow.FlowProject.pre` and :py:meth:`~flow.FlowProject.post` conditions, which allow us to define arbitrary workflows as a `directed acyclic graph <https://en.wikipedia.org/wiki/Directed_acyclic_graph>`__.

> A operation is only executed if **all** preconditions are met, and at *at least one* postcondition is not met.

Reviewer: ```suggestion
An operation is only executed if **all** preconditions are met, and at *at least one* postcondition is not met.
```
